src/rpc/RPC.cpp

```diff
@@ -118,31 +118,80 @@ make_error(Status const& status)
     json["type"] = "response";
     return json;
 }
-static std::unordered_map<std::string, std::function<Result(Context const&)>>
-    handlerTable{
-        {"account_channels", &doAccountChannels},
-        {"account_currencies", &doAccountCurrencies},
-        {"account_info", &doAccountInfo},
-        {"account_lines", &doAccountLines},
-        {"account_objects", &doAccountObjects},
-        {"account_offers", &doAccountOffers},
-        {"account_tx", &doAccountTx},
-        {"gateway_balances", &doGatewayBalances},
-        {"noripple_check", &doNoRippleCheck},
-        {"book_offers", &doBookOffers},
-        {"channel_authorize", &doChannelAuthorize},
-        {"channel_verify", &doChannelVerify},
-        {"ledger", &doLedger},
-        {"ledger_data", &doLedgerData},
-        {"ledger_entry", &doLedgerEntry},
-        {"ledger_range", &doLedgerRange},
-        {"ledger_data", &doLedgerData},
-        {"subscribe", &doSubscribe},
-        {"server_info", &doServerInfo},
-        {"unsubscribe", &doUnsubscribe},
-        {"tx", &doTx},
-        {"transaction_entry", &doTransactionEntry},
-        {"random", &doRandom}};
+
+using LimitRange = std::tuple<std::uint32_t, std::uint32_t, std::uint32_t>;
+using HandlerFunction = std::function<Result(Context const&)>;
+
+struct Handler
+{
+    std::string method;
+    std::function<Result(Context const&)> handler;
+    std::optional<LimitRange> limit;
+};
+
+class HandlerTable
+{
+    std::unordered_map<std::string, Handler> handlerMap_;
+
+public:
+    HandlerTable(std::initializer_list<Handler> handlers)
+    {
+        for (auto const& handler : handlers)
+        {
+            handlerMap_[handler.method] = std::move(handler);
+        }
+    }
+
+    bool
+    contains(std::string const& method)
+    {
+        return handlerMap_.contains(method);
+    }
+
+    std::optional<LimitRange>
+    getLimitRange(std::string const& command)
+    {
+        if (!handlerMap_.contains(command))
+            return {};
+
+        return handlerMap_[command].limit;
+    }
+
+    std::optional<HandlerFunction>
+    getHandler(std::string const& command)
+    {
+        if (!handlerMap_.contains(command))
+            return {};
+
+        return handlerMap_[command].handler;
+    }   
+};
+
+static HandlerTable handlerTable
+{
+    {"account_channels",   &doAccountChannels,   LimitRange{10, 50, 100}},
+    {"account_currencies", &doAccountCurrencies, {}},
+    {"account_info",       &doAccountInfo,       {}},
+    {"account_lines",      &doAccountLines,      LimitRange{10, 50, 100}},
+    {"account_objects",    &doAccountObjects,    LimitRange{10, 50, 100}},
+    {"account_offers",     &doAccountOffers,     LimitRange{10, 50, 100}},
+    {"account_tx",         &doAccountTx,         LimitRange{10, 20, 40}},
```

Comment:
I think we can bump the hi here. What's rippleds max?

Reply:
I'm unsure, but I don't think its super relevant here, since rippled uses a tx queue and we use coroutines.

We should tweak these based on latencies that we gather from graphana.

Reply:
The big gain here is that limits are now in a central location in the codebase, and are easily changed.